pkg/v1/webhooks/cluster_pause_webhook.go

```diff
@@ -52,17 +65,19 @@ func (wh *ClusterPause) Default(ctx context.Context, obj runtime.Object) error {
 		return err
 	}
 
-	if cluster.Spec.Topology != nil {
-		// Add pause to cluster if the topology.version changes
-		// The cluster pause state will be unset by ClusterBootstrap controller after it rolls out package updates
-		if currentCluster.Spec.Topology == nil || cluster.Spec.Topology.Version != currentCluster.Spec.Topology.Version {
-			cluster.Spec.Paused = true
-			if cluster.Annotations == nil {
-				cluster.Annotations = map[string]string{}
-			}
-			// Use the desired TKR version as label value, ClusterBootstrap will unset
-			cluster.Annotations[constants.ClusterPauseLabel] = cluster.Spec.Topology.Version
+	if currentCluster.Labels != nil {
+		currentTkrVersion, ok = currentCluster.Labels[v1alpha3.LabelTKR]
+	}
+
+	// Add pause to cluster if the cluster.Labels["run.tanzu.vmware.com/tkr"] changes
+	// The cluster pause state will be unset by ClusterBootstrap controller after it rolls out package updates
+	if currentCluster.Labels == nil || !ok || currentTkrVersion != tkrVersion {
```

Comment:
Could we remove `ok` check as we are already comparing `currentTkrVersion != tkrVersion`?